cmd/gardener-node-agent/app/bootstrappers/nodeagent_kubeconfig.go

```diff
@@ -0,0 +1,47 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package bootstrappers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/afero"
+	"k8s.io/client-go/rest"
+
+	"github.com/gardener/gardener/pkg/nodeagent"
+	nodeagentv1alpha1 "github.com/gardener/gardener/pkg/nodeagent/apis/config/v1alpha1"
+)
+
+// NodeAgentKubeconfig is a runnable to create the kubeconfig for the node-agent.
+type NodeAgentKubeconfig struct {
+	Log         logr.Logger
+	FS          afero.Afero
+	Cancel      context.CancelFunc
+	Config      *rest.Config
+	MachineName string
+}
+
+// Start performs creation of the gardener-node-agent kubeconfig.
+func (n *NodeAgentKubeconfig) Start(ctx context.Context) error {
+	ok, err := n.FS.Exists(nodeagentv1alpha1.KubeconfigFilePath)
+	if err != nil {
+		return err
+	}
+	if ok {
+		n.Log.Info("Kubeconfig file exists, skipping bootstrap")
+		return nil
+	}
+
+	n.Log.Info("Requesting kubeconfig for gardener-node agent")
+	if err := nodeagent.RequestAndStoreKubeconfig(ctx, n.Log, n.FS, n.Config, n.MachineName); err != nil {
+		return err
+	}
+
+	n.Log.Info("New kubeconfig written to disk successfully. Terminating gardener-node-agent")
```

Comment:
explain why GNA is terminated

Reply:
Currently, the `Start` method runs as a `BootstrapRunnable`. At this point the manager is already created using the bootstrap rest config. GNA should not run with this config, so it is restarting here.
It might make sense to not run this method as a `BootstrapRunnable`, but before the manager instance was created. Then GNA would not need to be restarted.

I'll check this in combination with the [refactoring of app.go](https://github.com/gardener/gardener/pull/10781#discussion_r1850717770).

Reply:
After I refactored `app.go` the file is gone.